framework/test/gtest/ResourceCompilerTest.cpp

```diff
@@ -38,44 +38,98 @@
+//uniqueTempdir is global as it needs to be a constant for all the tests
+TempDir uniqueTempdir = MakeUniqueTempDirectory();
+
 // The value returned by the resource compiler when manifest validation failed
 static const int BUNDLE_MANIFEST_VALIDATION_ERROR_CODE(2);
 
-/*
- * Execute a process with the given command line
- * @param executable absolute path to the executable to run along with
- *  and any command line parameters.
- * @return the executable's return code.
- */
-int runExecutable(const std::string& executable)
+class ResourceCompilerTest : public ::testing::Test
 {
-// WEXITSTATUS is only available on POSIX. Wrap std::system into a function
-// call so that there is consistent and uniform return codes on all platforms.
-#if defined US_PLATFORM_WINDOWS
-#  define WEXITSTATUS
-#endif
+public:
+  ResourceCompilerTest() = default;
+  ~ResourceCompilerTest() override = default;
 
-  int ret = std::system(executable.c_str());
+  void SetUp() override
+  {
+    rcbinpath = RCC_PATH;
+    /*
+     * If ResourceCompiler executable is not found, we can't run the tests, we
+     * mark it as a failure and exit
+     */
+    std::ifstream binf(rcbinpath.c_str());
+    ASSERT_TRUE(binf.good())
+      << "Cannot find usResourceCompiler executable:" << rcbinpath;
+
+    tempdir = uniqueTempdir.Path + DIR_SEP;
+  }
 
-  // WEXITSTATUS uses an old c-sytle cast
-  // clang-format off
-US_GCC_PUSH_DISABLE_WARNING(old-style-cast)
-  // clang-format on
-  return WEXITSTATUS(ret);
-  US_GCC_POP_WARNING
-}
+protected:
+  std::string rcbinpath;
+  std::string tempdir;
+
+  const std::string manifest_json = R"({
+    "bundle.symbolic_name" : "main",
+    "bundle.version" : "0.1.0",
+    "bundle.activator" : true
+    })";
+};
+
+// A zip file containing only a manifest.
+class ManifestZipFile
+{
+public:
+  // zip_file_name - zip file path
+  // manifest_json - contents of the manifest.json file
+  // bundle_name - name of the bundle
+  ManifestZipFile(const std::string& zip_file_name,
+                  const std::string& manifest_json,
+                  const std::string& bundle_name)
+  {
+    std::string archiveEntry(bundle_name + "/manifest.json");
+    mz_zip_archive zip;
+    memset(&zip, 0, sizeof(mz_zip_archive));
+
+    mz_zip_writer_init_file(&zip, zip_file_name.c_str(), 0);
+    mz_zip_writer_add_mem(&zip,
+                          archiveEntry.c_str(),
+                          manifest_json.c_str(),
+                          manifest_json.size() *
+                            sizeof(std::string::value_type),
+                          MZ_DEFAULT_COMPRESSION);
+    mz_zip_writer_finalize_archive(&zip);
+    mz_zip_writer_end(&zip);
+  }
+
+  ManifestZipFile(const ManifestZipFile&) = delete;
+  void operator=(const ManifestZipFile&) = delete;
+  ManifestZipFile(const ManifestZipFile&&) = delete;
+  ManifestZipFile& operator=(const ManifestZipFile&&) = delete;
+
+  ~ManifestZipFile() {}
```

Comment:
default?

Reply:
Did not change it from original test.
Also, default doesn't work here.